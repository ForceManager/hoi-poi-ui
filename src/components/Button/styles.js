export default (theme) => ({
    root: {
        position: 'relative',
        boxSizing: 'border-box',
        display: 'inline-block',
        minWidth: '100px',
        padding: '5px 16px',
        borderRadius: '100px',
        textAlign: 'center',
        cursor: 'pointer !important',
        '&:active': {
            ...theme.effects.buttonActive,
        },
        '&$fullWidth': {
            width: '100%',
        },
        '&$disabled': {
            backgroundPosition: '0 0 !important',
            opacity: 0.5,
            cursor: 'not-allowed !important',
            '&:hover': {
                backgroundPosition: '0 0 !important',
            },
            '& *': {
                cursor: 'not-allowed',
            },
        },
        '& *': {
            cursor: 'pointer',
        },
    },
    label: {
        isolate: true,
        display: 'inherit',
        alignItems: 'inherit',
        justifyContent: 'inherit',
        fontWeight: 500,
        transition: 'opacity .2s ease-in-out',
        ...theme.typography.button,
    },
    labelLoading: {
        isolate: true,
        position: 'relative',
        '& $label': {
            opacity: 0,
        },
        '& $loader': {
            opacity: 1,
        },
    },
    loader: {
        isolate: true,
        position: 'absolute',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        opacity: 0,
        transition: 'opacity .2s ease-in-out',
    },
    // Types
    outlined: {
        padding: '4px 16px',
        '&$primary': {
            background: 'transparent',
            backgroundColor: 'transparent',
            border: `1px solid ${theme.colors.primary}`,
            transition: `color .2s ease-in-out, box-shadow .2s ease-in-out, background-color .1s ease-in-out`,
            '&$disabled': {
                '&:hover': {
                    backgroundColor: 'transparent',
                    '& $label': {
                        color: theme.colors.primary,
                    },
                },
            },
            '& $label': {
                color: theme.colors.primary,
            },
            '&:hover': {
                backgroundColor: theme.colors.primary,
                '& $label': {
                    color: theme.colors.text.primary,
                },
            },
        },
        '&$danger': {
            background: 'transparent',
            backgroundColor: 'transparent',
            border: `1px solid ${theme.colors.danger}`,
            transition: `color .2s ease-in-out, box-shadow .2s ease-in-out, background-color .1s ease-in-out`,
            '& $label': {
                color: theme.colors.danger,
            },
            '&:hover': {
                backgroundColor: theme.colors.danger,
                '& $label': {
                    color: theme.colors.text.danger,
                },
            },
        },
        '&$small': {
            padding: '0 16px',
        },
        '&$big': {
            padding: '9px 16px',
        },
    },
    squared: {
        borderRadius: 2,
        padding: '4px 9px',
        minWidth: '0',
        background: 'transparent',
        backgroundColor: 'transparent',
        border: `1px solid ${theme.colors.placeholders}`,
        transition: `border-color .2s ease-in-out`,
        '&:hover': {
            border: `1px solid ${theme.colors.placeholdersHover}`,
            background: 'transparent',
            backgroundColor: 'transparent',
        },
        '&$disabled': {
            '&:hover': {
                border: `1px solid ${theme.colors.placeholders}`,
                backgroundColor: 'transparent',
            },
        },
        '&$white': {
            background: 'transparent',
            backgroundColor: 'transparent',
            '&$disabled': {
                '&:hover': {
                    backgroundColor: 'transparent',
                },
            },
            '&:hover': {
                background: 'transparent',
                backgroundColor: 'transparent',
            },
        },
        '&$primary': {
            background: 'transparent',
            backgroundColor: 'transparent',
            '&$disabled': {
                '&:hover': {
                    backgroundColor: 'transparent',
                    '& $label': {
                        color: theme.colors.primary,
                    },
                },
            },
            '& $label': {
                color: theme.colors.primary,
            },
        },
        '&$danger': {
            background: 'transparent',
            backgroundColor: 'transparent',
            '& $label': {
                color: theme.colors.danger,
            },
        },
        '&$small': {
            padding: '0 9px',
            minWidth: '0',
        },
        '&$big': {
            padding: '9px 9px',
            minWidth: '0',
        },
    },
    //Colors
    white: {
        backgroundColor: 'transparent',
        transition: `box-shadow .2s ease-in, background-color .1s ease-in-out`,
        '& $label': {
            color: theme.colors.text.default,
        },
        '&:hover': {
            backgroundColor: theme.colors.lightGray,
        },
    },
    primary: {
        background: `linear-gradient(to bottom, ${theme.colors.primary} 0%, ${
            theme.colors.primary
        } 50%, ${theme.colors.primaryDark} 100%)`,
        filter: `DXImageTransform.Microsoft.gradient( startColorstr='${
            theme.colors.primary
        }', endColorstr='${theme.colors.primaryDark}', GradientType=0 )`,
        backgroundSize: 'auto 200% !important',
        backgroundPosition: '100% 0 !important',
        transition: `box-shadow .2s ease-in, background-position .1s ease-in-out`,
        '& $label': {
            color: theme.colors.text.primary,
        },
        '&:hover': {
            backgroundPosition: '0 100% !important',
        },
    },
    danger: {
        background: `linear-gradient(to bottom, ${theme.colors.danger} 0%, ${
            theme.colors.danger
        } 50%, ${theme.colors.dangerDark} 100%)`,
        filter: `DXImageTransform.Microsoft.gradient( startColorstr='${
            theme.colors.danger
        }', endColorstr='${theme.colors.dangerDark}', GradientType=0 )`,
        backgroundSize: 'auto 200% !important',
        backgroundPosition: '100% 0 !important',
        transition: `box-shadow .2s ease-in, background-position .1s ease-in-out`,
        '& $label': {
            color: theme.colors.text.danger,
        },
        '&:hover': {
            backgroundPosition: '0 100% !important',
        },
    },
    // Sizes
    small: {
        minWidth: '60px',
        padding: '0px 10px',
    },
    big: {
        minWidth: '130px',
        padding: '10px 16px',
    },
    // States
    disabled: {},
    fullWidth: {},
});
