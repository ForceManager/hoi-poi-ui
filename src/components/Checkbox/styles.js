export default (theme) => ({
    root: {
        cursor: 'pointer',
    },
    input: {
        display: 'none',
    },
    disabled: {
        opacity: 0.5,
        cursor: 'not-allowed',
    },
});
