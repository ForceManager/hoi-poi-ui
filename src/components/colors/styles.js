export default (theme) => ({
    root: {},
    section: {
        paddingBottom: 20,
        marginBottom: 20,
        borderBottom: `1px solid ${theme.colors.grey[100]}`,
        '&:last-child': {
            borderBottom: 'none',
            marginBottom: 0,
        },
    },
    sectionTitle: {
        color: theme.colors.textLight.secondary,
        textTransform: 'capitalize',
        marginBottom: 12,
    },
    swatches: {
        display: 'flex',
        flexFlow: 'row wrap',
        padding: 12,
        gap: 12,
    },
    swatchesDark: {
        backgroundColor: theme.colors.background.black,
    },
    swatch: {
        display: 'flex',
        flexFlow: 'column nowrap',
        justifyContent: 'flex-end',
        alignItems: 'flex-end',
        width: 120,
        height: 60,
        padding: 8,
    },
    swatchTextDark: {
        color: theme.colors.primary.black,
    },
    swatchTextLight: {
        color: theme.colors.primary.white,
    },
});
