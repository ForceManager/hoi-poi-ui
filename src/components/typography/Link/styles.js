const typographies = [
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'subtitle1',
    'subtitle',
    'body1',
    'body',
    'button',
    'caption',
    'captionMedium',
    'badges',
    'overline',
];

export default (theme) => ({
    root: {
        display: 'inline-block',
        cursor: 'pointer !important',
        padding: 0,
        color: theme.colors.blue500,
        transition: 'color .1s ease-in-out',
        border: 'none',
        outline: 'none',
        background: 'transparent',
        textDecoration: 'none',
        '&:hover': {
            color: theme.colors.blue600,
        },
        '& *': {
            cursor: 'pointer',
        },
    },
    isDisabled: {
        opacity: 0.5,
        cursor: 'not-allowed !important',
        '&:hover': {
            backgroundPosition: '0 0 !important',
        },
        '& *': {
            cursor: 'not-allowed',
        },
    },
    ...typographies.reduce((obj, key) => ({ ...obj, [key]: { ...theme.typography[key] } }), {}),
    bold: { fontWeight: 500 },
    truncated: {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
        display: 'inherit',
    },
    underline: {
        '&:hover': {
            textDecoration: 'underline',
        },
    },
    primary: {
        color: theme.colors.orange500,
        '&:hover': {
            color: theme.colors.orange600,
        },
    },
});
