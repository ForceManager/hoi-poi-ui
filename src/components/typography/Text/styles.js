const typographies = [
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'subtitle1',
    'subtitle',
    'body1',
    'body',
    'button',
    'caption',
    'captionMedium',
    'badges',
    'overline',
];
export default (theme) => ({
    root: {
        display: 'inherit',
        color: theme.colors.neutral900,
    },
    ...typographies.reduce((obj, key) => ({ ...obj, [key]: { ...theme.typography[key] } }), {}),
    truncated: {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
    },
    bold: { fontWeight: 500 },
});
