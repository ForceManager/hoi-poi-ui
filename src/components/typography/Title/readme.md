Default:

```jsx
<Title>H1</Title>
```

### Component tree

---

-   root - root element
-   [Text](#/Typography?id=text)
