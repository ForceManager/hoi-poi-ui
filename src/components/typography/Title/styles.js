export default (theme) => ({
    root: {},
    textTitle: {
        fontFamily: 'Roboto',
        fontSize: '18px',
        fontWeight: 'normal',
        letterSpacing: '1.5px',
        color: theme.colors.text.black,
    },
});
