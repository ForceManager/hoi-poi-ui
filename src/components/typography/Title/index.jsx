import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'react-jss';
import classnames from 'classnames';
import styles from './styles';

import Text from '../Text';
import { getOverrides } from '../../../utils/overrides';

function Title({
    children,
    overrides: overridesProp,
    className: classNameProp,
    classes,
    ...props
}) {
    //Overrides
    const rootClassName = classnames(classes.root, classNameProp);

    const override = getOverrides(overridesProp, Title.overrides);

    const rootProps = {
        ...props,
        className: rootClassName,
        ...override.root,
    };

    return (
        <Text type="bold" {...rootProps}>
            <h1 className={classes.textTitle}>{children}</h1>
        </Text>
    );
}

Title.overrides = ['root'];

Title.defaultProps = {
    className: '',
    overrides: {},
};

Title.propTypes = {
    className: PropTypes.string,
    overrides: PropTypes.object,
    children: PropTypes.node.isRequired,
};

export default React.memo(withStyles(styles, { name: 'Title' })(Title));
