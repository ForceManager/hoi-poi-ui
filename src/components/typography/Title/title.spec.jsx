import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HoiPoiProvider } from '../../../utils/styles';
import Title from '../../typography/Title';

describe('Title', () => {
    test('Default Title', () => {
        const wrapper = mount(
            <HoiPoiProvider>
                <Title>H1</Title>
            </HoiPoiProvider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
