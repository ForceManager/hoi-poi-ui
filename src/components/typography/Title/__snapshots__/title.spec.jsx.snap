// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Default Title 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "black": "#262B34",
          "blackLight": "#5f656b",
          "disabledGrey": "#ced4d9",
          "green": "#86AD20",
          "greenLight": "#e7f1d5",
          "grey": "#697782",
          "greyLight": "#f0f3f5",
          "greySoft": "#788590",
          "lines": "#dce3eb",
          "link": "#216db3",
          "linkDark": "#174a79",
          "overlay": "rgba(33,33,33,0.55)",
          "placeholders": "#e1e4e7",
          "placeholdersHover": "#d0d4d8",
          "primary": "#FF8C00",
          "primaryDark": "#C87600",
          "primaryLight": "#fef4e6",
          "primarySoft": "#ff9d26",
          "red": "#f04540",
          "redDark": "#C43833",
          "redLight": "#FAE7E7",
          "redSoft": "#FF544F",
          "secondary": "#00A8Be",
          "secondaryGrey": "#e4eeff",
          "secondaryLight": "#D6EEF2",
          "secondarySoft": "#AADFE7",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "tableHeader": "#f6f7f9",
          "tableHeaderHover": "#EBECEF",
          "tableHover": "#FAFBFC",
          "text": Object {
            "black": "#3c4248",
            "danger": "white",
            "default": "#697782",
            "green": "#86AD20",
            "greySoft": "#788590",
            "primary": "white",
            "red": "#f04540",
            "secondary": "#00A8Be",
            "white": "#FFFFFF",
            "yellow": "#E9B200",
          },
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "white": "#FFFFFF",
          "yellow": "#E9B200",
          "yellowLight": "#FFF9DD",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 13,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "fontSizeSmall": 11,
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 2,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.0.0-alpha.16",
        }
      }
    >
      <JssContextSubscriber>
        <WithStyles(Title)
          className=""
          innerRef={null}
          jssContext={
            Object {
              "disableStylesGeneration": false,
              "jss": Jss {
                "generateId": [Function],
                "id": 2,
                "options": Object {
                  "Renderer": [Function],
                  "createGenerateId": [Function],
                  "plugins": Array [],
                },
                "plugins": PluginsRegistry {
                  "plugins": Object {
                    "external": Array [
                      Object {
                        "onCreateRule": [Function],
                        "onProcessStyle": [Function],
                        "onUpdate": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                        "onProcessRule": [Function],
                      },
                      Object {
                        "onProcessRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                        "onProcessRule": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessRule": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                    ],
                    "internal": Array [
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onCreateRule": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                    ],
                  },
                  "registry": Object {
                    "onChangeValue": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onCreateRule": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onProcessRule": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onProcessSheet": Array [],
                    "onProcessStyle": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onUpdate": Array [
                      [Function],
                    ],
                  },
                },
                "version": "10.0.0-alpha.16",
              },
              "managers": Object {
                "0": SheetsManager {
                  "length": 1,
                  "sheets": WeakMap {},
                },
                "1": SheetsManager {
                  "length": 1,
                  "sheets": WeakMap {},
                },
              },
              "sheetOptions": Object {
                "classNamePrefix": "HoiPoi__",
                "generateId": [Function],
              },
            }
          }
          overrides={Object {}}
          theme={
            Object {
              "colors": Object {
                "black": "#262B34",
                "blackLight": "#5f656b",
                "disabledGrey": "#ced4d9",
                "green": "#86AD20",
                "greenLight": "#e7f1d5",
                "grey": "#697782",
                "greyLight": "#f0f3f5",
                "greySoft": "#788590",
                "lines": "#dce3eb",
                "link": "#216db3",
                "linkDark": "#174a79",
                "overlay": "rgba(33,33,33,0.55)",
                "placeholders": "#e1e4e7",
                "placeholdersHover": "#d0d4d8",
                "primary": "#FF8C00",
                "primaryDark": "#C87600",
                "primaryLight": "#fef4e6",
                "primarySoft": "#ff9d26",
                "red": "#f04540",
                "redDark": "#C43833",
                "redLight": "#FAE7E7",
                "redSoft": "#FF544F",
                "secondary": "#00A8Be",
                "secondaryGrey": "#e4eeff",
                "secondaryLight": "#D6EEF2",
                "secondarySoft": "#AADFE7",
                "shadow": "rgba(0, 0, 0, 0.25)",
                "tableHeader": "#f6f7f9",
                "tableHeaderHover": "#EBECEF",
                "tableHover": "#FAFBFC",
                "text": Object {
                  "black": "#3c4248",
                  "danger": "white",
                  "default": "#697782",
                  "green": "#86AD20",
                  "greySoft": "#788590",
                  "primary": "white",
                  "red": "#f04540",
                  "secondary": "#00A8Be",
                  "white": "#FFFFFF",
                  "yellow": "#E9B200",
                },
                "transparentBlack": "rgba(38, 43, 52, 0.8)",
                "white": "#FFFFFF",
                "yellow": "#E9B200",
                "yellowLight": "#FFF9DD",
              },
              "effects": Object {
                "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
                "buttonActive": Object {
                  "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
                },
              },
              "typography": Object {
                "bigText": Object {
                  "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": 20,
                  "letterSpacing": 0.5,
                },
                "button": Object {
                  "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": 11,
                  "lineHeight": 1.5,
                },
                "defaultText": Object {
                  "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": 13,
                  "letterSpacing": 0.5,
                },
                "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSizeBase": 13,
                "fontSizeSmall": 11,
                "largeText": Object {
                  "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": 16,
                  "letterSpacing": 0.5,
                },
                "letterSpacing": 0.5,
                "lineHeightBase": 1.5,
                "smallText": Object {
                  "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": 11,
                  "letterSpacing": 0.5,
                },
              },
            }
          }
        >
          <Title
            className=""
            classes={
              Object {
                "root": "HoiPoi__Title__root-1-0",
                "textTitle": "HoiPoi__Title__textTitle-1-1",
              }
            }
            overrides={Object {}}
          >
            <JssContextSubscriber
              className="HoiPoi__Title__root-1-0"
              type="bold"
            >
              <WithStyles(Text)
                className="HoiPoi__Title__root-1-0"
                innerRef={null}
                isTruncated={false}
                jssContext={
                  Object {
                    "disableStylesGeneration": false,
                    "jss": Jss {
                      "generateId": [Function],
                      "id": 2,
                      "options": Object {
                        "Renderer": [Function],
                        "createGenerateId": [Function],
                        "plugins": Array [],
                      },
                      "plugins": PluginsRegistry {
                        "plugins": Object {
                          "external": Array [
                            Object {
                              "onCreateRule": [Function],
                              "onProcessStyle": [Function],
                              "onUpdate": [Function],
                            },
                            Object {
                              "onCreateRule": [Function],
                              "onProcessRule": [Function],
                            },
                            Object {
                              "onProcessRule": [Function],
                            },
                            Object {
                              "onCreateRule": [Function],
                              "onProcessRule": [Function],
                            },
                            Object {
                              "onChangeValue": [Function],
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onChangeValue": [Function],
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onChangeValue": [Function],
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onChangeValue": [Function],
                              "onProcessRule": [Function],
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                          ],
                          "internal": Array [
                            Object {
                              "onCreateRule": [Function],
                            },
                            Object {
                              "onCreateRule": [Function],
                            },
                            Object {
                              "onChangeValue": [Function],
                              "onCreateRule": [Function],
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onCreateRule": [Function],
                            },
                            Object {
                              "onCreateRule": [Function],
                            },
                            Object {
                              "onCreateRule": [Function],
                            },
                            Object {
                              "onCreateRule": [Function],
                            },
                          ],
                        },
                        "registry": Object {
                          "onChangeValue": Array [
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                          ],
                          "onCreateRule": Array [
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                          ],
                          "onProcessRule": Array [
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                          ],
                          "onProcessSheet": Array [],
                          "onProcessStyle": Array [
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                            [Function],
                          ],
                          "onUpdate": Array [
                            [Function],
                          ],
                        },
                      },
                      "version": "10.0.0-alpha.16",
                    },
                    "managers": Object {
                      "0": SheetsManager {
                        "length": 1,
                        "sheets": WeakMap {},
                      },
                      "1": SheetsManager {
                        "length": 1,
                        "sheets": WeakMap {},
                      },
                    },
                    "sheetOptions": Object {
                      "classNamePrefix": "HoiPoi__",
                      "generateId": [Function],
                    },
                  }
                }
                overrides={Object {}}
                size="medium"
                theme={
                  Object {
                    "colors": Object {
                      "black": "#262B34",
                      "blackLight": "#5f656b",
                      "disabledGrey": "#ced4d9",
                      "green": "#86AD20",
                      "greenLight": "#e7f1d5",
                      "grey": "#697782",
                      "greyLight": "#f0f3f5",
                      "greySoft": "#788590",
                      "lines": "#dce3eb",
                      "link": "#216db3",
                      "linkDark": "#174a79",
                      "overlay": "rgba(33,33,33,0.55)",
                      "placeholders": "#e1e4e7",
                      "placeholdersHover": "#d0d4d8",
                      "primary": "#FF8C00",
                      "primaryDark": "#C87600",
                      "primaryLight": "#fef4e6",
                      "primarySoft": "#ff9d26",
                      "red": "#f04540",
                      "redDark": "#C43833",
                      "redLight": "#FAE7E7",
                      "redSoft": "#FF544F",
                      "secondary": "#00A8Be",
                      "secondaryGrey": "#e4eeff",
                      "secondaryLight": "#D6EEF2",
                      "secondarySoft": "#AADFE7",
                      "shadow": "rgba(0, 0, 0, 0.25)",
                      "tableHeader": "#f6f7f9",
                      "tableHeaderHover": "#EBECEF",
                      "tableHover": "#FAFBFC",
                      "text": Object {
                        "black": "#3c4248",
                        "danger": "white",
                        "default": "#697782",
                        "green": "#86AD20",
                        "greySoft": "#788590",
                        "primary": "white",
                        "red": "#f04540",
                        "secondary": "#00A8Be",
                        "white": "#FFFFFF",
                        "yellow": "#E9B200",
                      },
                      "transparentBlack": "rgba(38, 43, 52, 0.8)",
                      "white": "#FFFFFF",
                      "yellow": "#E9B200",
                      "yellowLight": "#FFF9DD",
                    },
                    "effects": Object {
                      "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
                      "buttonActive": Object {
                        "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
                      },
                    },
                    "typography": Object {
                      "bigText": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                        "fontSize": 20,
                        "letterSpacing": 0.5,
                      },
                      "button": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                        "fontSize": 11,
                        "lineHeight": 1.5,
                      },
                      "defaultText": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                        "fontSize": 13,
                        "letterSpacing": 0.5,
                      },
                      "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                      "fontSizeBase": 13,
                      "fontSizeSmall": 11,
                      "largeText": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                        "fontSize": 16,
                        "letterSpacing": 0.5,
                      },
                      "letterSpacing": 0.5,
                      "lineHeightBase": 1.5,
                      "smallText": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                        "fontSize": 11,
                        "letterSpacing": 0.5,
                      },
                    },
                  }
                }
                type="bold"
              >
                <Text
                  className="HoiPoi__Title__root-1-0"
                  classes={
                    Object {
                      "big": "HoiPoi__Text__big-1-7",
                      "bold": "HoiPoi__Text__bold-1-8",
                      "large": "HoiPoi__Text__large-1-6",
                      "light": "HoiPoi__Text__light-1-10",
                      "medium": "HoiPoi__Text__medium-1-5",
                      "normal": "HoiPoi__Text__normal-1-9",
                      "root": "HoiPoi__Text__root-1-2",
                      "small": "HoiPoi__Text__small-1-4",
                      "truncated": "HoiPoi__Text__truncated-1-3",
                    }
                  }
                  isTruncated={false}
                  overrides={Object {}}
                  size="medium"
                  type="bold"
                >
                  <span
                    className="HoiPoi__Text__root-1-2 HoiPoi__Text__bold-1-8 HoiPoi__Text__medium-1-5 HoiPoi__Title__root-1-0"
                  >
                    <h1
                      className="HoiPoi__Title__textTitle-1-1"
                    >
                      H1
                    </h1>
                  </span>
                </Text>
              </WithStyles(Text)>
            </JssContextSubscriber>
          </Title>
        </WithStyles(Title)>
      </JssContextSubscriber>
    </JssProvider>
  </ThemeProvider>
</Component>
`;
